pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -70,6 +70,15 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	// update is not allowed to set TemplateGeneration
 	newDaemonSet.Spec.TemplateGeneration = oldDaemonSet.Spec.TemplateGeneration
 
+	// Update is not allowed to set Spec.Selector for all groups/versions except extensions/v1beta1.
+	// If RequestInfo is nil, it is better to revert to old behavior (i.e. allow update to set Spec.Selector)
+	// to prevent unintentionally breaking users who may rely on the old behavior.
+	// TODO(#50791): after v1beta1 is retired, move selector immutability check to validation codes
+	requestInfo, found := genericapirequest.RequestInfoFrom(ctx)
+	if found && !(requestInfo.APIGroup == "extensions" && requestInfo.APIVersion == "v1beta1") {
```

Comment:
Earlier, we agreed that we'd treat unknown versions (`default:` case) with the new behavior -- disallow mutation. Basically we want an explicit list of "legacy" versions (because we can make that list now and it will never change), and anything else defaults to the new behavior.

In this case, the the switch will only have one `case` and one `default` so it's basically an `if` statement. However, I agree that the general pattern of a switch makes sense; for example it will be clearer for Deployment where we'll list 2 legacy GroupVersions.